packages/python-core/superfunctions/oauth/flow.py

```
@@ -0,0 +1,848 @@
+            await self._emit_event(
+                {
+                    "name": "oauth.flow.callback.failed",
+                    "requestId": payload.request_id or _create_request_id(),
+                    "providerId": payload.provider_id or "",
+                    "at": _isoformat(now),
+                    "ok": False,
+                    "errorCode": wrapped.code,
+                    "details": wrapped.details,
+                }
+            )
+            raise wrapped
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)